From a35b03ee31a3d35f58bf994ea2597b38c9168dd1 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Sun, 17 Feb 2013 21:16:56 +0100
Subject: [PATCH 203/205] udev: usb_id: parse only 'size' bytes of the
 'descriptors' buffer (cherry picked from commit
 7962afbba9016ea03d9f2987fee341443fcde39d)

---
 src/udev/udev-builtin-usb_id.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev/udev-builtin-usb_id.c b/src/udev/udev-builtin-usb_id.c
index 5599958..b5167c3 100644
--- a/src/udev/udev-builtin-usb_id.c
+++ b/src/udev/udev-builtin-usb_id.c
@@ -189,7 +189,7 @@ static int dev_if_packed_info(struct udev_device *dev, char *ifs_str, size_t len
         pos = 0;
         strpos = 0;
         ifs_str[0] = '\0';
-        while (pos < sizeof(buf) && strpos+7 < len-2) {
+        while (pos < size && strpos+7 < len-2) {
                 struct usb_interface_descriptor *desc;
                 char if_str[8];
 
-- 
1.8.1.5

